01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb

```diff
@@ -2,29 +2,81 @@
 
 class OpenAddressing
   def initialize(size)
+    @items = Array.new(size)
   end
 
   def []=(key, value)
+    newIndex = index(key, size)
+    until @items[newIndex].nil? do
+      if next_open_index(newIndex) == -1
+        resize
+      else
+        newIndex = next_open_index(newIndex)
+      end
+    end
+
+    @items[newIndex] = Node.new(key, value)
   end
 
   def [](key)
+    newIndex = index(key, size)
+
+    if @items[newIndex].key != key
+      for i in 0..size do
+        newIndex = i if @items[i] && @items[i].key == key
+      end
+    end
+
+    @items[newIndex].value
```

Comment:
Something the test did not catch is that if the original index was not a match and the key is not actually found in the hash this will end up returning the wrong item. Try adding `expect(star_wars_movies["Star Wars: The Last Jedi"]).to be_nil` and see what happens.

Reply:
I entered
`      # Following line passes, strangely.
      expect(star_wars_movies["Star Wars: The Last Jedi"]).to be_nil`
on lines 59 & 60. The test passes.